pkg/services/thumbs/models.go

```diff
@@ -93,15 +50,17 @@ type crawlStatus struct {
 }
 
 type dashRenderer interface {
-	// Assumes you have already authenticated as admin
-	GetPreview(req *previewRequest) *previewResponse
 
 	// Assumes you have already authenticated as admin
-	Start(c *models.ReqContext, mode CrawlerMode, theme rendering.Theme) (crawlStatus, error)
+	Start(c *models.ReqContext, mode CrawlerMode, theme rendering.Theme, kind models.ThumbnailKind) (crawlStatus, error)
 
 	// Assumes you have already authenticated as admin
 	Stop() (crawlStatus, error)
 
 	// Assumes you have already authenticated as admin
 	Status() (crawlStatus, error)
+
+	SaveThumbnailFromFile(tempFilePath string, dashboardID int64, dashboardUID string, theme rendering.Theme, kind models.ThumbnailKind) (int64, error)
+	SaveThumbnailFromBytes(content []byte, mimeType string, dashboardID int64, dashboardUID string, theme rendering.Theme, kind models.ThumbnailKind) (int64, error)
+	GetMimeType(filePath string) string
```

Comment:
these feel like internal details that should not be exported or exposed on the interface.

The 'save' functions feel like they belong elsewhere, but assuming they are internal I think they are OK for now

Reply:
:+1: